packages/ra-ui-materialui/src/button/PrevNextButtons.tsx

```diff
@@ -0,0 +1,176 @@
+import ErrorIcon from '@mui/icons-material/Error';
+import { Link } from 'react-router-dom';
+import { CircularProgress, IconButton, SxProps, styled } from '@mui/material';
+import clsx from 'clsx';
+
+/**
+ * A component used to render the previous and next buttons in a Show or Edit view.
+ *
+ * The `<PrevNextButtons>` component render a navigation to move to the next or previous record
+ * of a resource, with the current index and the total records, in an edit or show view.
+ *
+ * It uses `usePrevNextController` to fetches the list of records.
+ *
+ * `<PrevNextButtons>` can be used anywhere a record context is provided
+ * (eg: often inside a `<Show>` or `<Edit>` component).
+ *
+ * @example <caption>move to edit view by default</caption>
+ * <ShowButton />
+ *
+ * @example <caption>move to show view</caption>
+ * <PrevNextButtons linkType="show" />
+ *
+ * @example <caption>share custom storeKey with other components</caption>
+ * <PrevNextButtons storeKey="listStoreKey" />
+ *
+ * @example <caption>limit the number of records to fetch</caption>
+ * <PrevNextButtons limit={500} />
+ *
+ * @example <caption>customize filters and sort order</caption>
+ * <PrevNextButtons
+ *     linkType="show"
+ *     sort={{
+ *         field: 'first_name',
+ *         order: 'DESC',
+ *     }}
+ *     filter={{ q: 'East a' }}
+ * />
+ *
+ * @example <caption>customize its style</caption>
+ * <PrevNextButtons
+ *     sx={{
+ *         color: 'blue',
+ *         '& .RaPrevNextButton-list': {
+ *             marginBottom: '20px',
+ *             color: 'red',
+ *         },
+ *     }}
+ * />
+ *
+ * @example <caption>in an edit view</caption>
+ * import * as React from "react";
+ * import { Edit, PrevNextButtons, ShowButton, SimpleForm, TopToolbar } from 'react-admin';
+ *
+ * const MyTopToolbar = ({ children }) => (
+ *     <TopToolbar>
+ *         {children}
+ *     </TopToolbar>
+ * );
+ *
+ * export const PostEdit = () => (
+ *      <Edit
+ *          actions={
+ *              <MyTopToolbar>
+ *                  <PrevNextButtons
+ *                      sort={{
+ *                          field: 'first_name',
+ *                          order: 'DESC',
+ *                      }}
+ *                      filter={{ q: 'East a' }}
+ *                  />
+ *                  <ShowButton />
+ *              </MyTopToolbar>
+ *          }
+ *      >
+ *          <SimpleForm>...</SimpleForm>
+ *      </Edit>
+ * );
+ */
+
+export const PrevNextButtons = <RecordType extends RaRecord = any>(
+    props: PrevNextButtonProps<RecordType>
+) => {
+    const { sx } = props;
+
+    const {
+        hasPrev,
+        hasNext,
+        navigateToNext,
+        navigateToPrev,
+        index,
+        total,
+        error,
+        isLoading,
+    } = usePrevNextController<RecordType>(props);
+
+    const translate = useTranslate();
+
+    if (isLoading) {
+        return <CircularProgress size={14} />;
```

Comment:
I'd use react-admin's `LinearProgress` instead, as it only shows the progress indicator after a 1 second delay.